### DISCUSSION TITLE
Why is there a large magic number here, rather than a more definitive termination condition?

### DISCUSSION THREAD

**Discussion by vsajip** [congo-cc/congo-parser-generator#133] [Q&A] [answered]
https://github.com/congo-cc/congo-parser-generator/blob/73c94d77cedf33a28da57d34a9ab4ee1d88ad8e9/src/templates/java/ParserProductions.java.ftl#L105

**Comment by revusky** [comment, marked as answer]
Well, I guess the reason is that this is how you do these things in FreeMarker. In Java, you'd just use while (true) {....} and under whatever conditions, you'd break out of the loop.  Arguably it's a flaw in the language. We could just have `#repeat ... /#repeat` and you could break out of the loop. Or optionally `#repeat until someCondition...`.
